File: Flutter/Vanilla/03_PigeonIntegrationTest/ios/Runner/NativeApiImplementation.swift
import UIKit
import Flutter

extension FlutterError: Error {}

class NativeApiImplementation: NativeApi {
  func getPlatformVersion() throws -> String {
    return  "iOS " + UIDevice.current.systemVersion
  }

  func getPlatformVersionAsync(completion: @escaping (Result<String, Error>) -> Void) {
    let version = try! self.getPlatformVersion()
    completion(.success(version))
  }
}

File: Flutter/Vanilla/01_Pigeon/android/app/src/main/kotlin/jp/taktak/demo/NativeApiImplementation.kt
package jp.taktak.demo

import NativeApi
import FlutterError

import androidx.annotation.NonNull
import io.flutter.embedding.android.FlutterActivity
import io.flutter.embedding.engine.FlutterEngine
import io.flutter.embedding.engine.plugins.FlutterPlugin

private class NativeApiImplementation: NativeApi {
  override fun getPlatformVersion(): String {
    return "Android ${android.os.Build.VERSION.RELEASE}"
  }
  override fun getPlatformVersionAsync(callback: (Result<String>) -> Unit) {
    callback(Result.success(this.getPlatformVersion()))
  }
}